src/udiskslinuxfilesystem.c

```diff
@@ -183,6 +186,19 @@ udisks_linux_filesystem_update (UDisksLinuxFilesystem  *filesystem,
   dev = udisks_linux_block_object_get_device_file (object);
   type = g_udev_device_get_property (device->udev_device, "ID_FS_TYPE");
   size = 0;
+
+  /* skip udisks_linux_filesystem_update when ata drive is inactive */
+  device = udisks_linux_block_object_get_device (object);
+  if (device != NULL)
+    {
+      if (g_udev_device_get_property_as_boolean (device->udev_device, "ID_ATA"))
+         {
+             awake = udisks_linux_drive_ata_is_active (device, &error, state);
```

Comment:
Code style - no need to use extra variable `awake`, you may test function result directly.

Reply:
Also leaking `error`.